collector/timex.go

```diff
@@ -163,6 +163,10 @@ func (c *timexCollector) Update(ch chan<- prometheus.Metric) error {
 
 	status, err := unix.Adjtimex(timex)
 	if err != nil {
+		if strings.Contains(err.Error(), "operation not permitted") {
```

Comment:
```suggestion
		if errors.Is(err, os.ErrPermission)
```

Reply:
I don't think this works. `ErrPermission` is "permission denied". This is a syscall error, not a filesystem open error. I went digging through the Go source and these errors are generated from a lookup table unlike the `os` ones. And the syscall package doesn't export these.

Reply:
Hrmm are you sure? Any other way around that except testing for the string? That seems rather brittle..

Reply:
Looking at the source at https://golang.org/src/syscall/syscall_unix.go?s=2831:2849#L126 I'd expect @roidelapluie's suggestion to work as os.ErrPermission is oserror.ErrPermission.

Reply:
https://godoc.org/golang.org/x/sys/unix

These calls return err == nil to indicate success; otherwise err represents an operating system error describing the failure and holds a value of type syscall.Errno.

https://golang.org/pkg/syscall/#Errno

Errno values can be tested against error values from the os package using errors.Is. For example:

```
_, _, err := syscall.Syscall(...)
if errors.Is(err, os.ErrNotExist) ...
```

Reply:
I have tested this and it works.

I have tested `if errors.Is(err, os.ErrPermission)` with Systemd (with seccomp enabled):

```
[Unit]
Description=Test Node Exporter

[Service]
ExecStart=/home/roidelapluie/go/src/github.com/prometheus/node_exporter/node_exporter --log.level=debug
User=roidelapluie
SystemCallFilter=~@clock
SystemCallErrorNumber=EPERM
```

```
Jan 23 14:06:38 oxygen node_exporter[14576]: level=debug ts=2021-01-23T13:06:38.190Z caller=timex.go:170 collector=timex msg="Not collecting timex metrics" err="operation not permitted"
```

Reply:
Update:


One shot command:

```
systemd-run --user --pty --same-dir --wait --collect --service-type=exec --property="SystemCallFilter=~@clock" --property="SystemCallErrorNumber=EPERM" ./node_exporter
```

Courtesy of https://blog.cloudflare.com/sandboxing-in-linux-with-zero-lines-of-code/

Reply:
Awesome, thanks for figuring that out.